### DISCUSSION TITLE
Misunderstanding ways to read ratelimit module limits

### DISCUSSION THREAD

**Discussion by barianiluca** [rspamd/rspamd#4753] [General]
With ratelimit module, if for a certain selector I want specify 50 mail to send by day (so in 24 hours), is that correct?

```
myrates = {
          selector = 'myselector';
          bucket = [
          {
                burst = 50;
                rate = "50 / 24h";
          }]
        }
```

The question rise because for certain of my users, this selector is correctly hit, and the rate correctly limited to 50 mail per days, but for others does not.
And for example if I try to create a loop script to test a domain in order to send something like 100 mail immediately, my testing script is catched by the above selector but is not stopped by that bucket. And all the 100 mail immediately relay out.

There is something in the middle that I lost?

**Comment by moisseev** [comment]
>  for certain of my users, this selector is correctly hit, and the rate correctly limited to 50 mail per days

Probably they have used up all burst capacity earlier.

> all the 100 mail immediately relay out.

That is expected, 50 messages of the burst + 50 messages of constant rate +/- some dynamic rate.

Apparently you need to set `burst = 0;` and

```hcl
max_rate_mult = 1;
max_bucket_mult = 1;
```

**Reply by barianiluca** [reply]
Nope, if I put:
```
burst = 0;
rate = "50 / 24h";
```
and  then:
```
max_rate_mult = 1;
max_bucket_mult = 1;
```
on ratelimit module configuration, no one email goes. Every mail is soft rejected. Maybe `burst = 1` make sense?

_to be sure that every counter is started from fresh, at every change of configuration I restart the rspamd and I cleanup the redis ratelimit keys._

**Reply by moisseev** [reply]
You can give it a try.

I in theory you just need to wait for half an hour after restart. The "50 / 24h" rate means 1 message per 28.8 seconds.
Sure with `burst = 1` it should pass 1 email immediately.
